board.go

```diff
@@ -0,0 +1,70 @@
+package honeycombio
+
+// Compile-time proof of interface implementation.
+var _ Boards = (*boards)(nil)
+
+// Boards describes (some of) the board related methods that Honeycomb supports.
+type Boards interface {
+	// List all boards present in this dataset.
```

Comment:
>This is not entirely true, since list does not use the dataset field. I created a new issue to look into this #2. This isn't blocking for this PR.

Good catch: a302e31798d61930ccec8c0a34c2eb39322718e1

I was actually considering opening a similar issue/having a discussion about that.